docs/cpp-api/bp-macros.md

```diff
@@ -0,0 +1,159 @@
+
+
+### `UE_COPY_STRUCT_PROPERTY_CUSTOM_BEGIN`: 
+Begins the process of copying an entire struct.
+- Param 1: The name, without quotes, of an ``FStructProperty`` that exists for this UFunction.
+- Param 2: A C++ compatible variable name for the property.
+
+
+### `UE_COPY_STRUCT_INNER_PROPERTY`: 
+Copies a property from within an `FStructProperty` into the already allocated params struct.
+- Param 1: The name, without quotes, of the `FStructProperty` supplied to `UE_COPY_STRUCT_PROPERTY_BEGIN`.
+- Param 2: The name, without quotes, of a property that exists in the supplied `FStructProperty`.
+- Param 3: The type that you want the underlying value to be copied as. For example, without quotes, "float" for `FFloatProperty`.
+- Param 4: The name of the C++ variable that you're copying.
+
+See: `AActor::K2_SetActorRotation`
+
+
+### `UE_COPY_STRUCT_INNER_PROPERTY_CUSTOM`:
+- Param 1: The name, without quotes, of the `FStructProperty` supplied to `UE_COPY_STRUCT_PROPERTY_BEGIN`.
+- Param 2: The name, without quotes, of a property that exists in the supplied `FStructProperty`.
+- Param 3: A C++ compatible variable name for the property.
+- Param 4: The type that you want the underlying value to be copied as. For example, without quotes, "float" for `FFloatProperty`.
+- Param 5: The name of the C++ variable that you're copying.
+
+
+### `UE_COPY_OUT_PROPERTY`: 
+Copies the out property of the supplied name from the params struct into the supplied C++ variable.
+
+This means the wrapper param (which is named the same as the property supplied) must be a reference, meaning suffixed with a "&".
+- Param 1: The name, without quotes, of a property that exists for this UFunction.
+- Param 2: The type that you want the underlying value to be copied as. For example, without quotes, "float" for `FFloatProperty`.
+
+See: `UGameplayStatics::FindNearestActor`
+
+
+### `UE_COPY_OUT_PROPERTY_CUSTOM`: 
+Copies the out property of the supplied name from the params struct into the supplied C++ variable.
+- Param 1: The name, without quotes, of a property that exists for this UFunction.
+- Param 2: A C++ compatible variable name for the property.
+- Param 3: The type that you want the underlying value to be copied as. For example, without quotes, "float" for `FFloatProperty`.
+
+This means the wrapper param (which is named the same as the property supplied) must be a reference, meaning suffixed with a "&".
+
+
+### `UE_COPY_VECTOR:` 
+Helper for copying an FVector. Must use `UE_COPY_STRUCT_PROPERTY_BEGIN` first.
+- Param 1: The C++ name, without quotes, of the FVector to copy from.
+- Param 2: The name, without quotes, of the FVector, same as supplied to `UE_COPY_STRUCT_PROPERTY_BEGIN`.
+
+
+### `UE_COPY_STL_VECTOR_AS_TARRAY`: 
+Helper for copying a TArray. (Marked in [BPMacros.hpp](https://github.com/Re-UE4SS/UEPseudo/blob/main/include/Unreal/BPMacros.hpp) as needing more testing, so I probably didn't test it thoroughly)
```

Comment:
Note for you left in, probably shouldn't be here or at least formatted in a more generic way that it needs testing.